app/controllers/classified_listings_controller.rb

```diff
@@ -10,13 +10,10 @@ def index
                            else
                              []
                            end
+    @displayed_classified_listing = ClassifiedListing.find_by!(category: params[:category], slug: params[:slug]) if params[:slug]
```

Comment:
Imo this should be `.find_by` and then a `not_found unless @displayed_classified_listing` after.